taui/src/index.js

```diff
@@ -6,14 +6,31 @@ import get from 'lodash/get'
 import React from 'react'
 import { connect } from 'react-redux'
 import { BrowserRouter, withRouter } from 'react-router-dom'
-import { withAuthenticator } from 'aws-amplify-react'
+import { SignIn } from 'aws-amplify-react'
 
 import actions from './actions'
 import awsmobile from './aws-exports'
 import Application from './components/application'
+import CustomSignIn from './components/custom-sign-in'
+import withAuthenticator from './components/with-authenticator'
 import reducers from './reducers'
 import * as select from './selectors'
 
+// custom Amplify component styling
+const CustomAuthenticatorTheme = {
```

Comment:
I dug into this a bit to try and understand the problem better and I don't think I have much to add. You've probably looked at this stuff already, but for @lederer 's benefit (and potential future readers), it looks like Amplify has [rolled their own CSS-in-JS solution](https://github.com/aws-amplify/amplify-js/blob/master/packages/aws-amplify-react/src/Amplify-UI/Amplify-UI-Components-React.jsx#L10) which they use pretty extensively, and it seems like that will make it awkward to put styling anywhere else as long as we're inheriting from Amplify. The two options I can think of quickly are to either have a single JS file for storing styles, which is what it looks like [Amplify does](https://github.com/aws-amplify/amplify-js/blob/master/packages/aws-amplify-react/src/Amplify-UI/Amplify-UI-Theme-Sample.jsx), or to try to add in some of our own classes that we can use for styling. @flibbertigibbet does that sound right to you? Are there any other options you've thought of?

@lederer do you have any ideas or opinions?

Reply:
The separate style JS file seems like a good compromise to me.

Reply:
Interesting. 

The _add in our own classes_ approach would fit the design team's usual practice of authoring [componentized SASS](https://github.com/azavea/facade/tree/develop/src/sass) that the build system transforms into a single `main.css` output file. 

Whereas, IIUC, the _JS theme file_ approach would forego all the value we get from SASS and autoprefixer, etc, and require authoring the equivalent of a `main.css` file directly (including prefixing, etc). I'm open to that if this is a small, one-off site. But if we expect it to grow, I'd prefer a scalable SASS-based solution.

Could there be a middle ground where we can transform a `main.css` output file into an Amplify theme file?

Reply:
Most of the styles are in a `main.css`. What we're discussing here are only the styles that override those that come from Amplify, which are the authentication components we're reusing (the sign-in page and header bar, primarily).

Reply:
There is a single `main.css` file for most of the styles. This is just to customize aspects of the authentication components (the sign-in and header bar, primarily), not an overall theme. I don't believe attempting to generate the Amplify JS theme overrides from CSS would be practical.

Reply:
Ah ok. Whew. 😅 

Customizing a theme file sounds fine for that.

Reply:
There's a section of the Cognito AWS console page for manually configuring per-domain/per user pool authentication component style overrides. I'm guessing this is related to why they've got JS-defined styles as they do, to support [that feature](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-app-ui-customization.html).

(I don't think remotely-managed styles would be useful for our purposes here, but it's interesting to know this feature exists.)

![image](https://user-images.githubusercontent.com/960264/52969200-806bfc80-337d-11e9-9ee4-1a4cb6cbeb1e.png)